odc/stats/plugins/gm.py

```diff
@@ -23,6 +23,7 @@ def __init__(
         cloud_classes: Tuple[str, ...],
         nodata_classes: Optional[Tuple[str, ...]] = None,
         filters: Optional[Iterable[Tuple[str, int]]] = None,
+        cloud_filters: Optional[Dict[str, Iterable[Tuple[str, int]]]]=None,
```

Comment:
Think we can have `Dict[Tuple[str], ...]` here, so that we will get rid of `cloud_classes` and `filters`? Also there is no need of line `80-91` or constructing a `tuple` in line `99` in this way.

Reply:
Hmm no, we have currently Dict[str, ..].  Dict[Tuple[str], ...] comes at the cost of ugly input:
 ('cloud', ): [['opening', 0], ['dilation', 5]],
 ('shadow', ): [['opening', 0], ['dilation', 5]]

I had kept the filters and cloud_classes for backwards compatibility but removed it since even if we need this feature, we can still achieve it by reformatting the input.

Reply:
Agreed, they’re indeed ugly once having values filled…